dandiapi/api/models/version.py

```diff
@@ -153,14 +157,63 @@ def citation(cls, metadata):
                 citation = f'{cl} ({year}) {name}. Online: {url}'
         return citation
 
+    @classmethod
+    def strip_metadata(cls, metadata):
+        """Strip away computed fields from a metadata dict."""
+        computed_fields = [
+            'name',
+            'identifier',
+            'version',
+            'id',
+            'url',
+            'citation',
+            'doi',
+            'datePublished',
+            'publishedBy',
+        ]
+        return {key: metadata[key] for key in metadata if key not in computed_fields}
+
     def _populate_metadata(self):
+        number_of_bytes = 0
+        number_of_files = 0
+        data_standards = set()
+        approaches = set()
+        measurement_techniques = set()
+        species = set()
+        # When running _populate_metadata on an unsaved Version, self.assets is not available.
+        # Only compute the asset-based properties if this Version has an id, which means it's saved.
+        if self.id:
+            number_of_bytes = (
+                self.assets.all().aggregate(size=models.Sum('blob__size'))['size'] or 0
+            )
+            number_of_files = self.assets.count()
+            for asset in self.assets.all():
+                metadata = asset.metadata.metadata
+                if 'dataStandard' in metadata:
+                    data_standards = data_standards.union(set(metadata['dataStandard']))
+                if 'approach' in metadata:
+                    approaches = approaches.union(set(metadata['approach']))
+                if 'measurementTechnique' in metadata:
+                    measurement_techniques = measurement_techniques.union(
+                        set(metadata['measurementTechnique'])
+                    )
+                if 'species' in metadata:
+                    species = species.union(set(metadata['species']))
```

Comment:
unfortunately many of these properties are not at the top level.  also this will need an update of all asset metadata as proposed in this PR: https://github.com/dandi/dandi-cli/pull/644/files (one of the reasons why we need to coordinate the full 0.4.0 upgrade)